pyfr/plugins/fluidforce.py

```diff
@@ -122,6 +160,29 @@ def __call__(self, intg):
             # Do the quadrature
             f += np.einsum('i...,ij,jik', qwts, p, norms)
 
+            if self.forcetype == 'viscous':
+                # Get operator and J^-T matrix
+                m4 = self._m4[etype]
+                rcpjact = self._rcpjact[etype, fidx]
+
+                # Transformed gradient at solution points
+                tduupts = np.dot(m4, uupts.reshape(nupts, -1))
+                tduupts = tduupts.reshape(ndims, nupts, nvars, -1)
+
+                # Physical gradient at solution points
+                duupts = np.einsum('ijkl,jkml->ikml', rcpjact, tduupts)
+
+                # Interpolate gradient to flux points
+                m5 = block_diag([m0]*ndims)
+                dufpts = np.dot(m5, duupts.reshape(ndims*nupts, -1))
+                dufpts = dufpts.reshape(ndims, nfpts, nvars, -1).swapaxes(1, 2)
```

Comment:
Anything below around ~1ms is noise and not really suitable for benchmarking purposes.